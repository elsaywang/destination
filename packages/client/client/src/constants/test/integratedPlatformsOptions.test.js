import {
    integratedPlatformsOptions,
    getIntegratedPlatformsOptionsTypeLabel,
} from '../integratedPlatformsOptions';

describe('integratedPlatformsOptions', () => {
    it('matches snapshot', () => {
        expect(integratedPlatformsOptions).toMatchSnapshot();
    });
    it('should return the correct label name given a valid integrated platform type value', () => {
        const validTypes = ['Integrated Platforms', 'People-Based', 'Device-Based'];
        validTypes.map(type => {
            const actual = getIntegratedPlatformsOptionsTypeLabel(type);
            const expected = integratedPlatformsOptions.find(({ label, value }) => value === type)
                .label;
            expect(actual).toEqual(expected);
        });
    });
    it('should return `` given an invalid integrated platform type value', () => {
        const actual = getIntegratedPlatformsOptionsTypeLabel('custom');
        const expected = ``;

        expect(actual).toEqual(expected);
    });
});
