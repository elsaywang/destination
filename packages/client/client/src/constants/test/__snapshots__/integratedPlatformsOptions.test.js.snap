// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integratedPlatformsOptions matches snapshot 1`] = `
Array [
  Object {
    "label": "ALL PLATFORMS",
    "value": "Integrated Platforms",
  },
  Object {
    "label": "PEOPLE-BASED",
    "value": "People-Based",
  },
  Object {
    "label": "DEVICE-BASED",
    "value": "Device-Based",
  },
]
`;
